### PR TITLE
Assembly length checks

### PR DESCRIPTION
**Pull request by hensha256**

### CODE REVIEW

File: test/libraries/CalldataDecoder.t.sol [hunk 1]
```diff
@@ -232,13 +289,58 @@ contract CalldataDecoderTest is Test {
...
+
+        bytes memory params = abi.encode(value);
+        bytes memory invalidParams = _removeFinalByte(params);
+        assertEq(invalidParams.length, params.length - 1);
+
+        vm.expectRevert(CalldataDecoder.SliceOutOfBounds.selector);
+        decoder.decodeUint256(invalidParams);
+    }
+
+    function test_fuzz_decodeCurrencyUint256AndBool(Currency _currency, uint256 _amount, bool _boolean) public {
```
Comment (hensha256) [inline test/libraries/CalldataDecoder.t.sol:322]: this test shouldve been here before... was just missing 🤷‍♀️

File: src/libraries/CalldataDecoder.sol [hunk 2]
```diff
@@ -136,6 +140,12 @@ library CalldataDecoder {
     {
         // ExactInputParams is a variable length struct so we just have to look up its location
         assembly ("memory-safe") {
+            // only safety checks for the minimum length, where path is empty
+            // 0xa0 = 5 * 0x20 -> 3 elements, path offset, and path length 0
+            if lt(params.length, 0xa0) {
```
[the remark this answers is not in the document]
Comment (hensha256) [reply src/libraries/CalldataDecoder.sol:145]: will check 👀

[the remark this answers is not in the document]
Comment (hensha256) [reply src/libraries/CalldataDecoder.sol:145]: update: `params.offset` holds the offset of where the content of the dynamic field starts, not the length

**Review by 0age — APPROVED** [review]

File: src/libraries/CalldataDecoder.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (0age) [reply src/libraries/CalldataDecoder.sol:186]: ah that's right! gains would be minimal if at all anyway 🙃

**Pull request merged**